### PR TITLE
updated the doc for setup

### PR DESCRIPTION
**Pull request by Albert-Gao**

### CODE REVIEW

File: docs/docs/intro.md [hunk 1]
```diff
@@ -100,6 +100,12 @@ If you use npm, this requires npm 8.3.0 or higher. You can install this with `np
 }
 ```
 
+enable the require context feature in metro by adding the following line to your `metro.config.js`
+
+```javascript title=metro.config.js
+defaultConfig.transformer.unstable_allowRequireContext = true
+```
```
Comment (EvanBacon) [inline docs/docs/intro.md:107]:
```suggestion
Ensure your `metro.config.js` extends the default [`expo/metro-config`](https://docs.expo.dev/guides/customizing-metro/).
```


This is enabled by default when using `expo/metro-config` in `expo@46.0.13`. We will be making other bundler changes as needed to support this project. We enforce that projects must extend `expo/metro-config` to use `expo-router`.

I'd rather not list out all of the configuration changes required behind the scenes to make this project work.

**Comment by Albert-Gao** [comment]
```bash
EXPO_ROUTER_APP_ROOT="/Users/albertgao/codes/self/teaming-cloud/tc-app" npx expo start --dev-client --clear
```

image

Do you think we need to add this env to the doc? I get it working by introducing this.

also, this is my metro config since I am not using monorepo

image

**Comment by EvanBacon** [comment]
The environment variable is injected by Expo CLI. I think it's ok to document adding an index.js file as a workaround: [example index.js](https://github.com/expo/examples/blob/master/with-router/index.js)

File: docs/docs/intro.md [hunk 2]
```diff
@@ -125,7 +127,7 @@ If you want to use the router with web, you'll need to enable Expo CLI's [experi
 Start the server with:
 
 ```
-npx expo start --clear
+EXPO_ROUTER_APP_ROOT="path/to/appRoot" npx expo start --clear
```
Comment (EvanBacon) [inline docs/docs/intro.md:130]: This should only ever be defined by [Expo CLI](https://github.com/expo/expo/pull/19231/files#diff-362d97681d5ead3b0b4b966473c0b148267e7dc68dba8af9734dfdfe3e0ad05dR56) and inlined by the babel plugin.

This logic is relatively new to React Native (using custom entry files on iOS is only supported in SDK 47) so I think it's fine to document the entry file.

**Comment by Albert-Gao** [comment]
I think I should close this one, since I believe the Expo v47 would make some current setup not-needed anymore.

**Pull request closed without merging**